src/test/common/steps/email-address-steps.js

```diff
@@ -1,7 +1,40 @@
-const { Then } = require('cucumber')
+const { When, Then } = require('cucumber')
+const { expect } = require('chai')
 
 const pages = require('./pages')
+const { enterEmailAddressAndSubmit, assertFieldErrorAndLinkTextPresentAndCorrect } = require('./common-steps')
+
+When(/^I enter a valid email address$/, async function () {
+  return enterEmailAddressAndSubmit()
+})
+
+When(/^I do not enter an email address$/, async function () {
+  return enterEmailAddressAndSubmit('')
+})
+
+When(/^I enter (.*) as my email address/, async function (emailAddress) {
+  // disable html5 form validation as this brings up a dialog box on chrome which interferes with the test.
+  await pages.driver.executeScript(setNoValidate)
+  return enterEmailAddressAndSubmit(emailAddress)
+})
 
 Then(/^I am shown the email address page$/, async function () {
   await pages.emailAddress.waitForPageLoad()
 })
+
+Then(/^I am informed that the email address is in the wrong format$/, async function () {
+  await assertFieldErrorAndLinkTextPresentAndCorrect(
+    pages.emailAddress.getEmailAddressFieldErrorId(),
+    pages.emailAddress.getEmailAddressLinkErrorCss(),
+    'Enter an email address in the correct format, like name@example.com')
+})
+
+Then(/^I see the email address (.*) in the textbox$/, async function (emailAddress) {
+  const enteredEmailAddress = await pages.emailAddress.getEmailAddressValue()
+  expect(enteredEmailAddress).to.be.equal(emailAddress)
+})
+
+const setNoValidate = () => {
```

Comment:
Can we have this at the top of the file please?

Reply:
All other step files has the steps followed by methods.